os/drivers/lcd/st7785.c

```diff
@@ -0,0 +1,343 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/lcd/lcd.h>
+#include <tinyara/lcd/lcd_dev.h>
+#include <tinyara/lcd/st7785.h>
+#include <tinyara/mipidsi/mipi_dsi.h>
+#include <tinyara/mipidsi/mipi_display.h>
+#include <debug.h>
+#include <assert.h>
+
+#define REGFLAG_DELAY                       0xFC
+#define REGFLAG_END_OF_TABLE                0xFD	// END OF REGISTERS MARKER
+
+#if !defined(CONFIG_LCD_MAXPOWER) || CONFIG_LCD_MAXPOWER < 1
+#define CONFIG_LCD_MAXPOWER 100
+#endif
+
+struct st7785_dev_s {
+	/* Publicly visible device structure */
+
+	struct lcd_dev_s dev;
+
+	FAR struct mipi_dsi_device *dsi_dev;
+
+	u8 *lcd_img_buffer[MAX_NO_PLANES];
+	//u8 *BackupLcdImgBuffer;
+	int fb_alloc_count;
+	uint8_t power;
+	struct st7785_config_s *config;
+};
+
+static struct st7785_dev_s g_lcdcdev;
+
+typedef struct lcm_setting_table {
+	u8 cmd;
+	u16 count;
+	u8 para_list[128];
+} lcm_setting_table_t;
+
+static lcm_setting_table_t st7785S_init_cmd_g[] = {	/* DCS Write Long */
```

Comment:
Typo: ST7785
The functions here in st7785.c are almost the same as st7701.c since both of them use MIPI protocol, thus I suggest to merge both into one, and provide a generic name for it, for example: st_7785_7789.c
And then, for the difference in between, such as the init table here, we could use 
```
#ifdef CONFIG_LCD_ST7701/CONFIG_LCDST7785
...
#endif
```